starfish/models/surfer_model.py

```diff
@@ -11,62 +11,84 @@
 from starfish import logger
 
 # service endpoint type name to use for this agent
-METADATA_MARKET_AGENT_ENDPOINT_NAME = 'metadata-storage'
-METADATA_MARKET_BASE_URI = '/api/v1/meta/data'
+SURFER_AGENT_ENDPOINT_NAME = 'metadata-storage'
+SURFER_BASE_URI = '/api/v1/meta/data'
 
-class MetadataAgentModel():
-    def __init__(self, ocean, did = None, authorization=None):
+class SurferModel():
+    _http_client = requests
+
+    def __init__(self, ocean, did=None, ddo=None, options=None):
         """init a standard ocan connection, with a given DID"""
         self._ocean = ocean
-        self._did = did
-        self._ddo = None
-        self._register_name = METADATA_MARKET_AGENT_ENDPOINT_NAME
 
-        # if DID then try to load in the linked DDO
-        if self._did:
-            self._ddo = self._resolve_did_to_ddo(self._did)
+        if options is None:
+            options = {}
+
 
         self._headers = {'content-type': 'application/json'}
-        if authorization:
+        if options.get('authorization'):
             self._headers['Authorization'] = f'Basic {authorization}'
 
+        if did is None or isinstance(did, str):
+            self._did = did
+        else:
+            raise ValueError('did must be a type string')
+
+        if ddo is None or isinstance(ddo, DDO) or isinstance(ddo, dict):
+            self._ddo = ddo
+        else:
+            raise ValueEror('ddo must be a DOD object or type dict')
+
+        # if DID then try to load in the linked DDO
+        if self._did and not self._ddo:
+            self._ddo = self._resolve_did_to_ddo(self._did)
+
     def register_asset(self, metadata):
         """
         Register an asset with the agent storage server
         :param metadata: metadata to write to the storage server
 
+        :return: A dict of the following items
+        [0] asset_id.
+        [1] did of the asset.
+        [2] metadata in text format, that was saved.
+
+        :type: dict
         """
         result = None
         metadata_text = json.dumps(metadata)
-        asset_id = MetadataMarketAgent.get_asset_id_from_metadata(metadata_text)
-        if self.save(asset_id, metadata_text):
+        asset_id = SurferModel.get_asset_id_from_metadata(metadata_text)
+        endpoint = self._get_endpoint(SURFER_AGENT_ENDPOINT_NAME)
+        saved_asset_id = self.save(asset_id, metadata_text, endpoint)
+        if asset_id == saved_asset_id:
             result = {
                 'asset_id': asset_id,
                 'did': f'{self._did}/{asset_id}',
                 'metadata_text': metadata_text,
             }
         return result
 
-    def save(self, asset_id, metadata_text):
+    def save(self, asset_id, metadata_text, endpoint):
```

Comment:
`save_metadata` or such would an intuitive name. Otherwise its not clear what  is being saved.

Reply:
yep, sounds good, since there will be other saves coming along